sdk/lib/cmlib/cmlib.h

```diff
@@ -202,13 +205,75 @@
+#define TAG_KCB            'bkMC'
+#define TAG_CMHIVE         'vHMC'
+#define TAG_CMSD           'DSMC'
+#define TAG_REGISTRY_STACK 'sRMC'
 
 #define CMAPI NTAPI
 
+//
+// Check Registry status type definition
+//
+typedef ULONG CM_CHECK_REGISTRY_STATUS;
+
+//
+// Check Registry flags
+//
+#define CM_CHECK_REGISTRY_DONT_PURGE_VOLATILES        0x0
+#define CM_CHECK_REGISTRY_PURGE_VOLATILES             0x2
+#define CM_CHECK_REGISTRY_BOOTLOADER_PURGE_VOLATILES  0x4
+#define CM_CHECK_REGISTRY_VALIDATE_HIVE               0x8
+
+//
+// Check Registry status codes
+//
+#define CM_CHECK_REGISTRY_GOOD                         0
+#define CM_CHECK_REGISTRY_SD_INVALID                   800
+#define CM_CHECK_REGISTRY_HIVE_CORRUPT_SIGNATURE       900
+#define CM_CHECK_REGISTRY_BIN_SIZE_OR_OFFSET_CORRUPT   905
+#define CM_CHECK_REGISTRY_BIN_SIGNATURE_HEADER_CORRUPT 1000
+#define CM_CHECK_REGISTRY_BAD_FREE_CELL                1005
+#define CM_CHECK_REGISTRY_BAD_ALLOC_CELL               1010
+#define CM_CHECK_REGISTRY_ALLOCATE_MEM_STACK_FAIL      2000
+#define CM_CHECK_REGISTRY_ROOT_CELL_NOT_FOUND          2005
+#define CM_CHECK_REGISTRY_BAD_LEXICOGRAPHICAL_ORDER    2010
+#define CM_CHECK_REGISTRY_NODE_NOT_FOUND               2015
+#define CM_CHECK_REGISTRY_SUBKEY_NOT_FOUND             2020
+#define CM_CHECK_REGISTRY_TREE_TOO_MANY_LEVELS         2025
+#define CM_CHECK_REGISTRY_KEY_CELL_NOT_ALLOCATED       3000
+#define CM_CHECK_REGISTRY_CELL_DATA_NOT_FOUND          3005
+#define CM_CHECK_REGISTRY_CELL_SIZE_NOT_SANE           3010
+#define CM_CHECK_REGISTRY_KEY_NAME_LENGTH_ZERO         3015
+#define CM_CHECK_REGISTRY_KEY_TOO_BIG_THAN_CELL        3020
+#define CM_CHECK_REGISTRY_BAD_KEY_NODE_PARENT          3025
+#define CM_CHECK_REGISTRY_BAD_KEY_NODE_SIGNATURE       3030
+#define CM_CHECK_REGISTRY_KEY_CLASS_UNALLOCATED        4000
+#define CM_CHECK_REGISTRY_VALUE_LIST_UNALLOCATED       5000
+#define CM_CHECK_REGISTRY_VALUE_LIST_DATA_NOT_FOUND    5005
+#define CM_CHECK_REGISTRY_VALUE_LIST_SIZE_NOT_SANE     5010
+#define CM_CHECK_REGISTRY_VALUE_CELL_NIL               6000
+#define CM_CHECK_REGISTRY_VALUE_CELL_UNALLOCATED       6005
+#define CM_CHECK_REGISTRY_VALUE_CELL_DATA_NOT_FOUND    6010
+#define CM_CHECK_REGISTRY_VALUE_CELL_SIZE_NOT_SANE     6015
+#define CM_CHECK_REGISTRY_CORRUPT_VALUE_DATA           6020
+#define CM_CHECK_REGISTRY_DATA_CELL_NOT_ALLOCATED      6025
+#define CM_CHECK_REGISTRY_BAD_KEY_VALUE_SIGNATURE      6030
+#define CM_CHECK_REGISTRY_STABLE_KEYS_ON_VOLATILE      7000
+#define CM_CHECK_REGISTRY_SUBKEYS_LIST_UNALLOCATED     7005
+#define CM_CHECK_REGISTRY_CORRUPT_SUBKEYS_INDEX        7010
+#define CM_CHECK_REGISTRY_BAD_SUBKEY_COUNT             7015
+#define CM_CHECK_REGISTRY_KEY_INDEX_CELL_UNALLOCATED   7020
+#define CM_CHECK_REGISTRY_CORRUPT_LEAF_ON_ROOT         7025
+#define CM_CHECK_REGISTRY_CORRUPT_LEAF_SIGNATURE       7030
+#define CM_CHECK_REGISTRY_CORRUPT_KEY_INDEX_SIGNATURE  7035
+
+//
+// Check Registry success macro
+//
```

Comment:
You do use CPP style comments here, although that is C code otherwise. I do **not** criticize that.
But it makes me wonder why you don't use the CPP style comments everywhere then? They would allow many of your comments to be written in a more compact manner.